plugins/llvm/llvm_binary.hpp

```diff
@@ -430,12 +430,13 @@ uint64_t image_entry(const COFFObjectFile* obj) {
 
 template <typename T>
 struct objectfile_image : image {
-    explicit objectfile_image(const T* obj)
+    explicit objectfile_image(const T* obj, std::unique_ptr<object::Binary> binary)
```

Comment:
Both parameters are referring to the same object under different types. The should be only one parameter of the most specific type, passing the same information twice is a bad practice. For example, here the prototype of the constructor should be `explicit objectfile_image(std::unique_ptr<T> binary)`